frame/ddc-staking/src/benchmarking.rs

```diff
@@ -0,0 +1,224 @@
+
+		let (origin_stash1, origin_controller1) = create_stash_controller_with_balance::<T>(
+			USER_SEED + 2,
+			origin_weight,
+		)?;
+
+		let (_origin_stash2, _origin_controller2) = create_stash_controller_with_balance::<T>(
+			USER_SEED + 3,
+			origin_weight,
+		)?;
+
+		// create an account with the worst case destination weight
+		let (_dest_stash1, _dest_controller1) = create_stash_controller_with_balance::<T>(
+			USER_SEED + 1,
+			origin_weight,
+		)?;
+
+		Ok(ListScenario { origin_stash1, origin_controller1, dest_weight: origin_weight })
+	}
+}
+
+const USER_SEED: u32 = 999666;
+
+benchmarks! {
+	bond {
+		let stash = create_funded_user::<T>("stash", USER_SEED, 100);
+		let controller = create_funded_user::<T>("controller", USER_SEED, 100);
+		let controller_lookup: <T::Lookup as StaticLookup>::Source
+			= T::Lookup::unlookup(controller.clone());
+		let amount = T::Currency::minimum_balance() * 10u32.into();
+		whitelist_account!(stash);
+	}: _(RawOrigin::Signed(stash.clone()), controller_lookup, amount)
+	verify {
+		assert!(Bonded::<T>::contains_key(stash));
+		assert!(Ledger::<T>::contains_key(controller));
+	}
+
+	bond_extra {
+		// clean up any existing state.
+		clear_storages_and_edges::<T>();
+
+		let origin_weight = MinStorageBond::<T>::get().max(T::Currency::minimum_balance());
+
+		// setup the worst case list scenario.
+
+		// the weight the nominator will start at.
+		let scenario = ListScenario::<T>::new(origin_weight)?;
+
+		let max_additional = scenario.dest_weight.clone() - origin_weight;
+
+		let stash = scenario.origin_stash1.clone();
+		let controller = scenario.origin_controller1.clone();
+		let original_bonded: BalanceOf<T>
+			= Ledger::<T>::get(&controller).map(|l| l.active).ok_or("ledger not created after")?;
+
+		T::Currency::deposit_into_existing(&stash, max_additional).unwrap();
+
+		whitelist_account!(stash);
+	}: _(RawOrigin::Signed(stash), max_additional)
+	verify {
+		let ledger = Ledger::<T>::get(&controller).ok_or("ledger not created after")?;
+		let new_bonded: BalanceOf<T> = ledger.active;
+	}
+
+	unbond {
+		// clean up any existing state.
+		clear_storages_and_edges::<T>();
+
+		// setup the worst case list scenario.
+		let total_issuance = T::Currency::total_issuance();
+		// the weight the nominator will start at. The value used here is expected to be
+		// significantly higher than the first position in a list (e.g. the first bag threshold).
+		let origin_weight = BalanceOf::<T>::try_from(952_994_955_240_703u128)
+			.map_err(|_| "balance expected to be a u128")
+			.unwrap();
+		let scenario = ListScenario::<T>::new(origin_weight)?;
+
+		let stash = scenario.origin_stash1.clone();
+		let controller = scenario.origin_controller1.clone();
+		let amount = origin_weight - scenario.dest_weight.clone();
+		let ledger = Ledger::<T>::get(&controller).ok_or("ledger not created before")?;
+		let original_bonded: BalanceOf<T> = ledger.active;
+
+		whitelist_account!(controller);
```

Comment:
Why are we doing `whitelist_account` everywhere?

Reply:
https://substrate.stackexchange.com/questions/4207/benchmarking-whitelist